eZ/Publish/API/Repository/Tests/LocationServiceTest.php

```diff
@@ -3238,6 +3238,126 @@ public function testMoveSubtreeUpdatesPathIdentificationString(): void
         );
     }
 
+    /**
+     * Test that is_visible is set correct for children when moving a location where a child is hidden by content (not by location).
+     *
+     * @covers \eZ\Publish\API\Repository\LocationService::moveSubtree
+     * depends eZ\Publish\API\Repository\Tests\LocationServiceTest::testLoadLocation
+     */
+    public function testMoveSubtreeKeepsContentHiddenOnChildrenAndParent(): void
+    {
+        $repository = $this->getRepository();
+
+        $mediaLocationId = $this->generateId('location', 43);
+
+        $locationService = $repository->getLocationService();
+        $contentService = $repository->getContentService();
+
+        $sourceFolderContent = $this->publishContentWithParentLocation('SourceFolder', $mediaLocationId); // media/SourceFolder
+        $subFolderContent1 = $this->publishContentWithParentLocation('subFolderContent1', $sourceFolderContent->contentInfo->mainLocationId);
+        $subFolderContent2 = $this->publishContentWithParentLocation('subFolderContent2', $sourceFolderContent->contentInfo->mainLocationId);
+        $targetFolderContent = $this->publishContentWithParentLocation('targetFolder', $mediaLocationId); // media/SourceFolder
+
+        $contentService->hideContent($subFolderContent1->contentInfo);
+
+        // Move media/SourceFolder to media/TargetFolder/
+        $locationService->moveSubtree(
+            $sourceFolderContent->contentInfo->getMainLocation(),
+            $targetFolderContent->contentInfo->getMainLocation()
+        );
+
+        $movedLocation = $locationService->loadLocation($sourceFolderContent->contentInfo->mainLocationId);
+        $newParentLocation = $locationService->loadLocation($targetFolderContent->contentInfo->mainLocationId);
+
+        // Assert Moved Location remains visible ( only child is hidden )
+        $this->assertPropertiesCorrect(
+            [
+                'hidden' => false,
+                'invisible' => false,
+                'depth' => $newParentLocation->depth + 1,
+                'parentLocationId' => $newParentLocation->id,
+                'pathString' => $newParentLocation->pathString . $this->parseId('location', $movedLocation->id) . '/',
+            ],
+            $movedLocation
+        );
+        self::assertFalse($movedLocation->getContentInfo()->isHidden);
+
+        // Assert children of Moved location
+        foreach ([$subFolderContent1->contentInfo->getMainLocation(), $subFolderContent2->contentInfo->getMainLocation()] as $childLocation) {
```

Comment:
CS please extract the array to a separate variable for a readability

Reply:
Fixed in I have anyway now removed the line in a99fb158a9447ae40c7638048cd981a209de12d1